src/lib/comms/include/sol-gatt.h

```diff
@@ -0,0 +1,256 @@
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_gatt_unregister_attributes(struct sol_gatt_attr *attrs, uint16_t len);
+
+/**
+ * @brief Reads the value from a attribute
+ *
+ * This only really makes sense for Characteristics and Descriptors,
+ * the Read Long attribute procedure will be performed if it's supported.
+ *
+ * @param conn Connection to a remote device
+ * @param attr Attribute to be read
+ * @param cb Callback to be called when the operation is finished
+ * @param user_data User pointer to be passed to the callback
+ */
+int sol_gatt_read_attr(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+		       void (*cb)(void *user_data, bool success,
+				  struct sol_gatt_attr *attr,
+				  struct sol_buffer *buf),
+		       const void *user_data);
+
+/**
+ * @brief Writes the value to a attribute
+ *
+ * This only really makes sense for Characteristics and Descriptors,
+ * the Write Long attribute procedure will be performed if it's supported.
+ *
+ * @param conn Connection to a remote device
+ * @param attr Attribute to be written
+ * @param cb Callback to be called when the operation is finished
+ * @param user_data User pointer to be passed to the callback
+ */
+int sol_gatt_write_attr(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+			struct sol_buffer *buf,
+			void (*cb)(void *user_data, bool success,
+				   struct sol_gatt_attr *attr)
+			const void *user_data);
+
+/**
+ * @brief Discover attributes by type, restricted by UUID and parent attribute.
+ *
+ * Discover attributes belonging to a parent attribute, for example,
+ * discover all the characteristics under a service or all the
+ * descriptors under a characteristic, matching uuid (if provided).
+ */
+int sol_gatt_discover(struct sol_bt_conn *conn, enum sol_gatt_attr_type type,
+		      const struct sol_gatt_attr *parent,
+		      const struct sol_bt_uuid *uuid,
+		      bool (*func)(void *user_data, struct sol_bt_conn *conn,
+				   const struct sol_gatt_attr *attr),
+		      const void *user_data);
+
+/**
+ * Registers a callback to be called when a notification/indication is received.
+ *
+ * It will also try to enable indications/nofications writing to the
+ * CCC attribute if that descriptor exists.
+ */
+int sol_gatt_subscribe(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+		       bool (*cb)(void *user_data, struct sol_gatt_attr *attr,
+				  const uint8_t *value, uint16_t len),
+		       const void *user_data);
+
+/**
+ * Unregisters from receiving notifications indications
+ */
+int sol_gatt_unsubscribe(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+		       bool (*cb)(void *user_data, struct sol_gatt_attr *attr,
+				  const uint8_t *value, uint16_t len));
+
+/**
+ * Sends an indication to the device represented by @a conn, if @a conn is NULL, send to all
+ * devices that have registered themselves via the CCC attribute.
+ *
+ * The value to be indicated is the value retrieved by calling @a read callback of the attribute.
```

Comment:
since this relates to `sol_gatt_subscribe()`, cross link them with `@see funcname()`